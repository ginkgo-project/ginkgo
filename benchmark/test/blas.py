#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"n": 100}]'],
    expected_stdout="blas.simple.stdout",
    expected_stderr="blas.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="blas.simple.stdout",
    expected_stderr="blas.simple.stderr",
    stdin='[{"n": 100}]',
)

# file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.blas.json")],
    expected_stdout="blas.simple.stdout",
    expected_stderr="blas.simple.stderr",
    stdin='[{"n": 100}]',
)

# profiler annotations
test_framework.compare_output(
    ["-input", '[{"n": 100}]', "-profile", "-profiler_hook", "debug"],
    expected_stdout="blas.profile.stdout",
    expected_stderr="blas.profile.stderr",
    stdin='[{"n": 100}]',
)
