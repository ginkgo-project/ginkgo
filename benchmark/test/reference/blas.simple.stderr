This is Ginkgo 1.7.0 (develop)
    running with core module 1.7.0 (develop)
Running on reference(0)
Running with 2 warm iterations and 10 running iterations
The random seed for right hand sides is 42
The operations are copy,axpy,scal
Running test case
{
    "n": 100,
    "blas": {}
}
Current state:
[
    {
        "n": 100,
        "blas": {
            "copy": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "completed": true
            }
        }
    }
]
Current state:
[
    {
        "n": 100,
        "blas": {
            "copy": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "axpy": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "completed": true
            }
        }
    }
]
Current state:
[
    {
        "n": 100,
        "blas": {
            "copy": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "axpy": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "completed": true
            },
            "scal": {
                "time": 1.0,
                "flops": 1.0,
                "bandwidth": 1.0,
                "repetitions": 10,
                "completed": true
            }
        }
    }
]
